### DISCUSSION TITLE
FR: Section setting to auto update slugs if entry title changes

### DISCUSSION THREAD

**Discussion by richhayler** [craftcms/cms#8645] [Ideas]
### Description
Would be great if there was an option to "auto update slug on entry title change". For some sections it would be great for consistency that the slug reflects the entry title in cases where the entry title may be changed after the initial entry save.

Although when updating an entry you can edit the slug manually that's annoying from a user perspective and it's also not possible in cases when editing an entry via a modal edit screen within another entry edit screen.

**Comment by brandonkelly** [comment]
Craft doesn’t do this because changing the slug is likely to break URLs or [reference tags](https://docs.craftcms.com/v3/reference-tags.html) to the entry elsewhere. Though if you have [Retour](https://plugins.craftcms.com/retour) installed you’re at least covered on the URL front, as that automatically creates redirects for you when the slug changes.

Worth noting that from the Edit Entry page, you can have Craft auto-generate a new slug based on the title by deleting the current Slug value. If that’s empty when you click Save, Craft will generate a new slug on the back end for you.

**Comment by richhayler** [comment]
Thanks for the explanation Brandon. Good tip deleting the slug to auto generate a new one. And I didn't know about Retour automatically creating redirects 👍

If a site however doesn't use reference tags and an author has the ability to manually change a slug after it has initially been generated then I can see a use case for having an option to handle this automatically. I do get that implementing this as an option could create issues for someone who doesn't fully appreciate the implications as you've pointed out above and presumably the reason for it working the way it does currently. Will give this some more thought and consideration - thanks
